frontend/src/routes/ClusterManagement/Clusters/Clusters.tsx

```diff
@@ -420,10 +420,14 @@ export function ClustersTable(props: {
                                             isValidError: errorIsNot([ResourceErrorCode.NotFound]),
                                         })
                                     },
-                                    isDisabled: !tableActionRbacValues['cluster.destroy'],
-                                    tooltip: !tableActionRbacValues['cluster.destroy']
-                                        ? t('common:rbac.unauthorized')
-                                        : '',
+                                    isDisabled: cluster.isManaged
+                                        ? !tableActionRbacValues['cluster.destroy']
+                                        : !tableActionRbacValues['cluster.destroy.detached'],
+                                    tooltip:
+                                        (cluster.isManaged && !tableActionRbacValues['cluster.destroy']) ||
+                                        !tableActionRbacValues['cluster.destroy.detached']
+                                            ? t('common:rbac.unauthorized')
+                                            : '',
```

Comment:
@jamestalton This logic could use a second pair of eyes. It performed well against my manual tests, I am wondering if it looks good to you.